src/Picqer/Financials/Exact/Query/Findable.php

```diff
@@ -169,13 +179,9 @@ public function collectionFromResult($result, array $headers = [])
                 $nextResult = [$nextResult];
             }
 
-            $result = array_merge($result, $nextResult);
-        }
-        $collection = [];
-        foreach ($result as $r) {
-            $collection[] = new static($this->connection(), $r);
+            foreach ($nextResult as $row) {
+                yield new self($this->connection(), $row);
```

Comment:
```suggestion
                yield new static($this->connection(), $row);
```
I guess we should keep using the late static binding method to avoid issues in the inheritance tree.